feat(builder-page): Add personal builder profile page

All-Khwarizmi opened the pull request:
## Description
![Capture d’écran 2025-01-08 à 05 42 10](https://github.com/user-attachments/assets/5c5a91a2-fabc-4048-a2ce-02fe07c7b3a5)

Added personal builder profile page featuring:
- Introduction & social links
- Journey timeline
- Focus areas
- Languages 

## Additional Information
- [x] First contribution to batch repository
- [x] Following issue requirements

## Related Issues
#11 

Your ENS/address: 0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E

phipsae commented on packages/nextjs/app/builders/0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E/page.tsx:
```diff
@@ -0,0 +1,134 @@
+import { Address } from "~~/components/scaffold-eth";
+
+export default function Page() {
+  const BUILDER_INFO = {
+    name: "Jason Suárez",
+    role: "Smart Contract | Frontend Developer",
+    address: "0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E",
+    avatar: "https://ik.imagekit.io/montresor/9.png?updatedAt=1730742966131",
+    tagline: "Securing the blockchain, one contract at a time",
+    languages: ["🇫🇷", "🇩🇴", "🇬🇧"],
+    links: {
+      bg: "https://app.buidlguidl.com/builders/0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E",
+      github: "https://github.com/All-Khwarizmi",
+      twitter: "https://twitter.com/swarecito",
+      telegram: "https://t.me/swarecito",
+    },
+    focus: {
+      building: "Smart Contracts",
+      learning: "DeFi security and cross-chain mechanics",
+      contributing: "BuildGuild batch12 and open source",
+    },
+    journey: [
+      {
+        year: "2023",
+        milestone: "Frontend Developer",
+        detail: "Building modern web applications",
+      },
+      {
+        year: "2024",
+        milestone: "Blockchain Development",
+        detail: "Smart contract development & security",
+      },
+      {
+        year: "2025",
+        milestone: "BuildGuild #12",
+        detail: "Contributing to Web3 ecosystem",
+      },
+    ],
+  };
```
You consider to place the BUILDER_INFO out of the component as a good practice.

phipsae reviewed (changes requested):
Thanks for your nice personal page!

You paid attention to mobile responsivness and dark/light mode. 
Also the PR description is very good! Good job! 

Below you'll find some little comments.

All-Khwarizmi commented:
@phipsae I've implemented the requested changes:
- Added NextPage type
- Changed to Next.js Image component (and updated the next.config accordingly)
- Moved BUILDER_INFO to module scope and used made it read only
- Renamed component for clarity

Thank you for the review and the detailed suggestions. Learning a lot already 🚀

phipsae reviewed (changes requested):
Thanks for the changes!

Just found a little thing. Then we are ready to merge.

phipsae commented on packages/nextjs/app/builders/0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E/page.tsx:
```diff
@@ -0,0 +1,143 @@
+    },
+  ],
+} as const;
+export const JasonBuilderProfile: NextPage = () => {
```
To ensure compatibility with the Next.js deployment on vercel, we need to include a default export at the bottom of the code, such as export default JasonBuilderProfile.

All-Khwarizmi replied:
I had the `default` export actually. 

I had to remove the `named` export for this to work.

phipsae reviewed (changes requested):
I just discovered another small issue that causes the following error during deployment to Vercel:

`Type error: Page "app/builders/0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E/page.tsx" does not match the required types of a Next.js Page.
  "JasonBuilderProfile" is not a valid Page export field.`

phipsae commented:
I am a bit confused, why is @NikolaiL pushing his page to this PR? 
@All-Khwarizmi: If you want to have the latest version of the app, pls merge `main` into it.  
We don't have have his profile page in this PR.

Let me know when you have any questions or need support!

All-Khwarizmi commented:
> I am a bit confused, why is @NikolaiL pushing his page to this PR? @All-Khwarizmi: If you want to have the latest version of the app, pls merge `main` into it. We don't have have his profile page in this PR.
> 
> Let me know when you have any questions or need support!

Could it be related to the fact that I rebase instead of merging to have the latest changes ? 

That's what I usually do when working on a feature for some time. I feel it helps avoiding conflicts but maybe that's something that should be handled in the PR?

All-Khwarizmi commented on packages/nextjs/app/builders/0x7d971C39bb700AcEc20879D46f092dC0DB1dbF9E/page.tsx:
```diff
@@ -0,0 +1,143 @@
+                width={128}
+                height={128}
+                className="rounded-full object-cover"
+              />
```
Solved! 

Added the `priority` property to improve the LCP loading.

**For educational purposes** here's the Next.js docs on the Image component [priority](https://nextjs.org/learn-pages-router/seo/web-performance/lcp) property and a related [explanation](https://nextjs.org/learn-pages-router/seo/web-performance/lcp) of the LCP (and core web vitals) .

phipsae commented:
> > I am a bit confused, why is @NikolaiL pushing his page to this PR? @All-Khwarizmi: If you want to have the latest version of the app, pls merge `main` into it. We don't have have his profile page in this PR.
> > Let me know when you have any questions or need support!
> 
> Could it be related to the fact that I rebase instead of merging to have the latest changes ?
> 
> That's what I usually do when working on a feature for some time. I feel it helps avoiding conflicts but maybe that's something that should be handled in the PR?

Ah true :)

As a reviewer, it’s confusing to see all the changes and files that have already been merged into `main` appearing again in this PR. This makes it harder to focus on the actual changes being introduced. Since `main` is already in production and your PR will be merged into it, I believe the correct approach would be to merge the latest `main` into your branch. This makes the review process clearer in this context clearer. 

Does this makes sense?

All-Khwarizmi commented:
> > > I am a bit confused, why is @NikolaiL pushing his page to this PR? @All-Khwarizmi: If you want to have the latest version of the app, pls merge `main` into it. We don't have have his profile page in this PR.
> > > Let me know when you have any questions or need support!
> > 
> > 
> > Could it be related to the fact that I rebase instead of merging to have the latest changes ?
> > That's what I usually do when working on a feature for some time. I feel it helps avoiding conflicts but maybe that's something that should be handled in the PR?
> 
> Ah true :)
> 
> As a reviewer, it’s confusing to see all the changes and files that have already been merged into `main` appearing again in this PR. This makes it harder to focus on the actual changes being introduced. Since `main` is already in production and your PR will be merged into it, I believe the correct approach would be to merge the latest `main` into your branch. This makes the review process clearer in this context clearer.
> 
> Does this makes sense?

Yes makes perfect sense! 

This the perfect example for someone wanting to argue in favor of merging over rebasing.

phipsae commented:
Thanks a lot for the nice personal page and its merged.
